iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/LiteralImplTest.java

```diff
@@ -27,13 +27,14 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.sonar.iac.common.testing.TextRangeAssert.assertTextRange;
+import static org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar.REGULAR_STRING_LITERAL;
 import static org.sonar.iac.docker.tree.impl.DockerTestUtils.parse;
 
 class LiteralImplTest {
 
   @Test
```

Comment:
I would add few tests cases like `"foo$bar5a"` or `"foo$$bar"`, and maybe have a full test case with assertion for the last one to understand how it is parsed.
Also maybe also add a test with an escape quote : `"foo\"bar"`
BTW I just tested this : `LABEL lab=my$$test`, and it seems to be equivalent of `LABEL lab=mytest`
So I expect that we will have a lot of funny use case to handle later <3

Reply:
^this comment should be for `ExpandableStringLiteralImplTest`